### PR TITLE
Activity versions

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
Now that the data migrations have been deployed, this PR routes the activity version from the front end (`promptStep.tsx`) through to the FeedbackController, so that it can create FeedbackHistories with the correct activity version.

The UI form to manage the versioning will be shipped in a following PR.

## WHY
Part of the evidence activity versioning project. 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Evidence-Activity-Versioning-f3e8454cd5f14b2ebda15dddd796fdf5

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | (N/A or Yes)

### CODE REVIEW

File: services/QuillLMS/app/models/feedback_history.rb [hunk 1]
```diff
@@ -134,7 +134,15 @@ def most_recent_rating
   end
 
   # TODO: consider making this a background job.
-  def self.save_feedback(feedback_hash_raw, entry, prompt_id, activity_session_uid, attempt, api_metadata=nil)
+  def self.save_feedback(
```
Comment (brendanshean) [inline services/QuillLMS/app/models/feedback_history.rb:137]: Thank you for breaking this up into multiple lines.

Comment (brendanshean) [inline services/QuillLMS/app/models/feedback_history.rb:137]: Thank you for breaking this up into multiple lines.

File: services/QuillLMS/client/app/bundles/Evidence/components/studentView/container.tsx [hunk 2]
```diff
@@ -286,7 +286,8 @@ export const StudentViewContainer = ({ dispatch, session, isTurk, location, acti
         promptText,
         attempt,
         previousFeedback,
-        callback: submitResponseCallback
+        callback: submitResponseCallback,
+        activityVersion: currentActivity?.version
```
Comment (brendanshean) [inline services/QuillLMS/client/app/bundles/Evidence/components/studentView/container.tsx:290]: I did a double take on `?.`.  Cool!

**Review by anathomical — APPROVED** [review]
Looks good to me.

File: services/QuillLMS/engines/evidence/app/controllers/evidence/feedback_controller.rb [hunk 3]
```diff
@@ -101,7 +103,15 @@ def spelling
       feedback = feedback.except(:api)
       attempt = params[:attempt] || 0
 
-      Evidence.feedback_history_class.save_feedback(feedback, @entry, @prompt.id, @session_id, attempt, api_metadata)
+      Evidence.feedback_history_class.save_feedback(**{
```
Comment (brendanshean) [inline services/QuillLMS/engines/evidence/app/controllers/evidence/feedback_controller.rb:106]: I might be misidentifying the positional arguments deprecation but the `save_feedback` calls feels like [scenario 1](https://www.bigbinary.com/blog/ruby-2-7-deprecates-conversion-of-keyword-arguments)

For this and the other occurrences, are double-splat and braces needed here?  What about omitting them altogether:

```
FeedbackHistory.save_feedback(
  feedback_hash_raw: feedback_hash_with_blank_metadata,
  entry: entry,
  prompt_id: prompt_id,
  activity_session_uid: activity_session_uid,
  attempt: attempt,
  activity_version: 0
)
```

**Review by brendanshean — APPROVED** [review]
Nice work on this.  One minor question about positional arguments, but nothing that should hold this up.  I like the batch cleanup too.

**Pull request merged**